File: zul/src/org/zkoss/zul/api/Treecol.java
/* Treecol.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

/**
 * A treecol.
 * <p>
 * Default {@link #getZclass}: z-treecol (since 5.0.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Treecol extends org.zkoss.zul.impl.api.HeaderElement {

	/**
	 * Returns the tree that it belongs to.
	 */
	public org.zkoss.zul.api.Tree getTreeApi();

	/**
	 * Returns the maximal length of each item's label.
	 */
	public int getMaxlength();

	/**
	 * Sets the maximal length of each item's label.
	 */
	public void setMaxlength(int maxlength);

	/**
	 * Returns the column index, starting from 0.
	 */
	public int getColumnIndex();

}

File: zul/src/archive/web/js/zul/sel/mold/treecol.js
/* treecol.js

	Purpose:
		
	Description:
		
	History:
		Wed Jun 10 15:31:55     2009, Created by jumperchen

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	var zcls = this.getZclass();
	out.push('<th', this.domAttrs_(), '><div id="', this.uuid, '-cave" class="',
			zcls, '-cnt">', this.domContent_());
	for (var w = this.firstChild; w; w = w.nextSibling)
		w.redraw(out);
	out.push('</div></th>');	
}
